Fixes HitTest for Align

Nash0x7E2 opened the pull request:
## Description
This PR solves the problem of foreground widgets not receiving pointer events when used in an Align with a width factor overlap. 

## Related Issue
- Fixes https://github.com/flutter/flutter/issues/60844

### Pre-patch:
![pre align patch](https://user-images.githubusercontent.com/25674767/86523886-320bb080-be41-11ea-8bad-cfbb3b249d89.gif)

### Post-Patch
![post align patch](https://user-images.githubusercontent.com/25674767/86523895-451e8080-be41-11ea-9c3e-0eba2c0fa283.gif)


## Tests

- Hit test on Align 

## Checklist

- [ x ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ x ] I signed the [CLA].
- [ x ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ x ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ x ] I updated/added relevant documentation (doc comments with `///`).
- [ x ] All existing and new tests are passing.
- [ x ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ x ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [ x ] No, no existing tests failed, so this is *not* a breaking change.

Nash0x7E2 committed:
Fixes hit test on RenderAligningShiftedBox

Nash0x7E2 committed:
remove extra space

Nash0x7E2 committed:
add widget test

Piinks reviewed (commented):
Hello again @Nash0x7E2! Thank you for another contribution! 😄  It looks like this is causing several test failures. Can you take a look?

Nash0x7E2 commented:
Hi again @Piinks 😀  Sure! The analysis is failing due to an unused import (my bad 😬) however looking at the logs, I am not sure why the framework widget tests are failing. Maybe I am missing something?

Nash0x7E2 committed:
remove unused import

Nash0x7E2 commented:
@Piinks I've updated this PR and removed the unused import. I looked at the failure logs for Linux and windows then tried running these tests locally and they passed. I am not entirely sure what's happening there :/

Piinks reviewed (commented):
Hey @Nash0x7E2 I checked out your branch and updated it with the latest from master, and it looks like this is breaking some existing tests based on the Center widget extending the Align class. It appears that there are cases where this hit testing implementation is not desirable, perhaps we can explore this and make it more configurable? 
cc'ing @goderbauer and @dkwingsmt who look to be a bit more familiar in this area based on the tests that are breaking.

goderbauer commented:
Thanks for the contribution. Unfortunately, this will not work: As soon as the RenderPositionedBox is wrapped in another RenderObject whose size is smaller then the RenderPositionedBox it will not receive hits, example:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(AlignGesture());
}

class AlignGesture extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Container(
        color: Colors.blue,
        child: Align(
          widthFactor: 0.5,
          child: Container(
            height: 100,
            width: 100,
            color: Colors.green,
            child: GestureDetector(
              onTap: () {
                print('tapped');
              },
            ),
          ),
        ),
      ),
    );
  }
}
```

![Screen Shot 2020-07-13 at 4 29 22 PM](https://user-images.githubusercontent.com/1227763/87363464-15990380-c526-11ea-9c41-37b97031172b.png)

When you hit the green area that's not inside the blue, the `onTap` handler will not fire.

This is by design. For performance reasons, a render object only receives hits inside the bounds of its parent. If you draw outside of the bounds of your parent, you will not be able to receive hit tests.

goderbauer commented:
I am going to close this PR since this is working as intended.
